cachi2/core/resolver.py

```diff
@@ -30,16 +30,24 @@ def resolve_packages(request: Request) -> RequestOutput:
     """
     Resolve all packages specified in a request.
 
-    This function performs the operations in a working copy of the source directory to avoid
-    modifications.
+    This function performs the operations in a working copy of the source directory in case
+    a package manager that can make unwanted modifications will be used.
     """
+    if request.yarn_packages:
+        return _call_on_source_dir_work_copy(request, _resolve_packages)
+    else:
+        return _resolve_packages(request)
+
+
+def _call_on_source_dir_work_copy(request: Request, handler: Handler) -> RequestOutput:
+    """Call the handler function on work copy of a request's source dir."""
```

Comment:
I went with the latter. I kind of liked the previous version as it was compact and clear, but probably because I wrote it, so it was not confusing to me :grimacing:

Reply:
Every review is subjective. The only reason I commented on this hunk was that whenever I see a callable being passed around I get triggered, because IMO callbacks have a very specific use case based on architectural design, e.g. plugin implementation, drivers, etc. I simply didn't think we needed that approach in your scenario.